VogtObserver/RandomNumberGenerator.cs

```diff
@@ -31,7 +41,19 @@ public void Remove(IObserver o)
 
         public int GetRandomInt()
         {
-            return _random.Next(1, 1000);
+            int number = 0;
+            try
+            {
+                number = _random.Next(1, 1000);
+                NumberChanged(number);
```

Comment:
Switch this to Notify(number) and get rid of the NumberChanged() function altogether. There is no reason to have NumberChanged -- its only calling another function